classes/class-wc-connect-shipping-label.php

```diff
@@ -0,0 +1,310 @@
+			$formData = [];
+			$contents = array();
+
+			foreach( $order->get_items() as $item ) {
+				$product = $order->get_product_from_item( $item );
+				$height = 0;
+				$length = 0;
+				$weight = $product->get_weight();
+				$width = 0;
+
+				if ( $product->has_dimensions() ) {
+					$height = $product->get_height();
+					$length = $product->get_length();
+					$width  = $product->get_width();
+				}
+
+				$contents[] = array(
+					'height' => $height,
+					'product_id' => $item['product_id'],
+					'length' => $length,
+					'quantity' => $item['qty'],
+					'weight' => $weight,
+					'width' => $width
+				);
+			}
+			$formData[ 'cart' ] = $contents;
+
+			foreach( $order->get_address( 'shipping' ) as $key => $value ) {
+				$formData[ 'orig_' . $key ] = '';
+				$formData[ 'dest_' . $key ] = $value;
+			}
+
+			return $formData;
+		}
+
+		protected function get_form_layout() {
+			$layout = [];
+
+			$address_fields = array(
+				array(
+					'key' => 'first_name',
+				),
+				array(
+					'key' => 'last_name',
+				),
+				array(
+					'key' => 'company',
+				),
+				array(
+					'key' => 'address_1',
+				),
+				array(
+					'key' => 'address_2',
+				),
+				array(
+					'key' => 'city',
+				),
+				array(
+					'key' => 'state',
+				),
+				array(
+					'key' => 'postcode',
+				),
+				array(
+					'key' => 'country',
+				),
+			);
+			$address_summary = '{first_name} {last_name}\\n{address_1} {address_2}\\n{city}, {postcode} {state}, {country}';
+
+			$orig_address_fields = $address_fields;
+			$dest_address_fields = $address_fields;
+			foreach( $address_fields as $index => $_ ) {
+				$orig_address_fields[ $index ][ 'key' ] = 'orig_' . $address_fields[ $index ][ 'key' ];
+				$dest_address_fields[ $index ][ 'key' ] = 'dest_' . $address_fields[ $index ][ 'key' ];
+			}
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Origin',
+				'title' => 'Generate shipping label: Address verification (origin)',
+				'description' => 'asdfghjkjhgfdsaASDFGHJJHGFDSA',
+				'items' => array( array( 'items' => $orig_address_fields ) ),
+				'bypass_suggestion_flag' => 'orig_bypass_suggestion',
+				'summary' => str_replace( '{', '{orig_', $address_summary ),
+				'sugggestion_hint' => 'vnkdjfitysncgkgse,jhxccvnaluerghsldjfvbzdj',
+			);
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Destination',
+				'title' => 'Generate shipping label: Address verification',
+				'description' => 'asdfghjkjhgfdsaASDFGHJJHGFDSA',
+				'items' => array( array( 'items' => $dest_address_fields ) ),
+				'bypass_suggestion_flag' => 'dest_bypass_suggestion',
+				'summary' => str_replace( '{', '{dest_', $address_summary ),
+				'sugggestion_hint' => 'qwertyuiopqwertyuiopqwerty uiopqwertyuiopqwertyuiop',
+			);
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Packages',
+				'title' => 'Shopping cart contents',
+				'items' => array(
+					array(
+						'items' => array(
+							array(
```

Comment:
So. Much. Nesting.

An "step" needs to have an array of "items", and each of those "items" must be a "group" (like a `fieldset` for example), and each of those groups will have an array of "items", each one of them will be an actual field.

Another approach could be to treat an "step" as a special type of group... I don't know, I have mixed feelings either way.